### PR TITLE
feat: add scheduled pipeline for k8s autodiscovery test suite

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It adds a multibranch pipeline for running the "Kubernetes Autodiscover" test suite for the supported maintenance branches in a nightly manner.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests for the CLI, and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [x] @jsoriano, which slack channel should we use to notify errors or green builds?

## Related issues

- Relates elastic/beats#25307

### CODE REVIEW

File: .ci/jobs/e2e-testing-k8s-autodiscovery-daily-mbp.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    name: e2e-tests/e2e-testing-k8s-autodiscovery-daily-mbp
+    display-name: End-2-End tests for Observability K8S Autodiscovery Pipeline
+    description: Run E2E K8S Autodiscovery test suite daily, including maintenance branches
+    view: E2E
+    project-type: multibranch
+    script-path: .ci/e2eTestingK8SAutodiscoveryDaily.groovy
+    scm:
+      - github:
+          branch-discovery: no-pr
+          head-filter-regex: '(master|7\.x|7\.13\.x|7\.12\.x)'
```
Comment (mdelapenya) [inline .ci/jobs/e2e-testing-k8s-autodiscovery-daily-mbp.yml:12]: Including the upcoming 7.13.x. If we should wait for 7.14 artifacts to appear, let's add this commit:
```suggestion
          head-filter-regex: '(master|7\.x|7\.12\.x|7\.11\.x)'
```

**Review by cachedout — APPROVED** [review]

**Review by jsoriano — APPROVED** [review]
Thanks!

**Comment by jsoriano** [comment]
> which slack channel should we use to notify errors or green builds?

I guess that `beats-build`.

File: .ci/e2eTestingK8SAutodiscoveryDaily.groovy [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  }
+  stages {
+    stage('Run Tests') {
+      steps {
+        build(job: "e2e-tests/e2e-testing-mbp/${env.JOB_BASE_NAME}",
+          parameters: [
+            booleanParam(name: 'forceSkipGitChecks', value: true),
+            booleanParam(name: 'forceSkipPresubmit', value: true),
+            string(name: 'runTestsSuites', value: 'kubernetes-autodiscover'),
+            string(name: 'SLACK_CHANNEL', value: "integrations"),
```
Comment (mdelapenya) [inline .ci/e2eTestingK8SAutodiscoveryDaily.groovy:48]:
```suggestion
            string(name: 'SLACK_CHANNEL', value: "beats-build"),
```

Comment (jsoriano) [reply .ci/e2eTestingK8SAutodiscoveryDaily.groovy]: Well, seeing now the change `integrations` could be ok too, so not sure, as you prefer.

Comment (mdelapenya) [reply .ci/e2eTestingK8SAutodiscoveryDaily.groovy]: I'd recommend using the channel your team is going to pay attention in the case of a failure 😄

Comment (jsoriano) [reply .ci/e2eTestingK8SAutodiscoveryDaily.groovy]: Then it is probably better to use `integrations` :D `beats-build` has many things.

File: .ci/e2eTestingK8SAutodiscoveryDaily.groovy [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  }
+  stages {
+    stage('Run Tests') {
+      steps {
+        build(job: "e2e-tests/e2e-testing-mbp/${env.JOB_BASE_NAME}",
+          parameters: [
+            booleanParam(name: 'forceSkipGitChecks', value: true),
+            booleanParam(name: 'forceSkipPresubmit', value: true),
+            string(name: 'runTestsSuites', value: 'kubernetes-autodiscover'),
+            string(name: 'SLACK_CHANNEL', value: "beats-build"),
```
Comment (mdelapenya) [inline .ci/e2eTestingK8SAutodiscoveryDaily.groovy:48]:
```suggestion
            string(name: 'SLACK_CHANNEL', value: "integrations"),
```

**Review by jsoriano — APPROVED** [review]

**Pull request merged**